.github/workflows/ci.yaml

```
@@ -0,0 +1,118 @@
+---
+name: 'CI'
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  checks: write  # For publishing test results
+
+jobs:
+  run-unit-tests:
+    name: 'Run Unit Tests'
+    runs-on: macos-14
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v6
+        with:
+          submodules: true
+
+      - name: Setup environment
+        uses: ./.github/actions/setup
+        env:
+          VGS_VAULT_ID: ${{ secrets.VGS_VAULT_ID }}
+
+      - name: Run Detekt
+        run: ./gradlew vgsshow:detekt --continue
+
+      - name: Run Unit Tests
+        run: ./gradlew vgsshow:testDebugUnitTest --continue
+
+      - name: (Compass) Generate Code Coverage Report
+        if: github.ref == 'refs/heads/main'
+        run: ./gradlew vgsshow:koverHtmlReportDebug
+
+      - name: (Compass) Extract Code Coverage
+        id: coverage
+        if: github.ref == 'refs/heads/main'
+        run: |
+          COVERAGE=$(sed -n '51p' vgsshow/build/reports/kover/htmlDebug/index.html)
```

```suggestion
          ./gradlew vgsshow:koverHtmlReportDebug
          COVERAGE=$(sed -n '51p' vgsshow/build/reports/kover/htmlDebug/index.html)
```

Since you need a condition, it's a bit simpler to have a single job